src/basic/fd-util.c

```diff
@@ -761,9 +761,10 @@ int rearrange_stdio(int original_input_fd, int original_output_fd, int original_
 }
 
 int fd_reopen(int fd, int flags) {
-        int new_fd, r;
+        int r;
 
         assert(fd >= 0 || fd == AT_FDCWD);
+        assert(!FLAGS_SET(flags, O_CREAT));
```

Comment:
Instead, how about to silently drop the flag `flags &= ~O_CREAT;` to make `fd_reopen()` work gracefully ?
Though, not a strong opinion. Feel free to ignore.

Reply:
I considered that, but in fd_reopen_condition we acquire the flags of original fd and compare it with the new flags to decide whether to reopen. If O_CREAT is passed as new flag, we may unnecessarily reopen it.

This can be resolved through adding some logic there too, but I think it's not worth it. I'd like to make sure that the callers of fd_reopen don't pass O_CREAT in the first place.